### PR TITLE
Add env var pass through for allauth provider secrets

### PR DESCRIPTION
**Pull request by agjohnson**
This has a matching community PR to support settings these from the host. This is similar to the pattern we're using for Stripe secrets

### CODE REVIEW

File: dockerfiles/docker-compose.yml [hunk 1]
```diff
@@ -195,6 +210,21 @@ services:
       - RTD_PRODUCTION_DOMAIN
       - RTD_LOGGING_LEVEL
       - RTD_DJANGO_DEBUG
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_CLIENT_ID
```
Comment (stsewd) [inline dockerfiles/docker-compose.yml:213]: Don't think the builders need to have access to those

File: dockerfiles/docker-compose.yml [hunk 2]
```diff
@@ -109,6 +109,21 @@ services:
       - RTD_STRIPE_SECRET
       - RTD_STRIPE_PUBLISHABLE
       - RTD_DJSTRIPE_WEBHOOK_SECRET
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_CLIENT_ID
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_KEY
```
Comment (stsewd) [inline dockerfiles/docker-compose.yml:113]: We can omit the _KEY from all providers

File: dockerfiles/docker-compose.yml (hunk 1 above)
Comment (agjohnson) [reply dockerfiles/docker-compose.yml:213]: Yeah I leaned towards no too. I'll remove these and the `_key` settings.

File: dockerfiles/docker-compose.yml [hunk 3]
```diff
@@ -109,6 +109,16 @@ services:
...
       - RTD_STRIPE_PUBLISHABLE
       - RTD_DJSTRIPE_WEBHOOK_SECRET
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_CLIENT_ID
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_SECRET
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITLAB_CLIENT_ID
+      - RTD_SOCIALACCOUNT_PROVIDERS_GITLAB_SECRET
+      - RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_OAUTH2_CLIENT_ID
+      - RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_OAUTH2_SECRET
+      - RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_CLIENT_ID
+      - RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_SECRET
```
Comment (stsewd) [inline dockerfiles/docker-compose.yml:119]: We aren't actively using these (are deprecated by allauth too). https://github.com/readthedocs/readthedocs.org/issues/11162

```suggestion
```

**Review by stsewd — APPROVED** [review]

**Pull request merged**